packages/wrangler/src/api/pages/publish.tsx

```diff
@@ -243,13 +248,55 @@ export async function publish({
 	 * When using a _worker.js file, the entire /functions directory is ignored
 	 * – this includes its routing and middleware characteristics.
 	 */
-	if (_workerJS) {
+	if (_workerJSDirectory) {
+		const traverseModuleGraphResult = await traverseModuleGraph(
+			{
+				file: resolvePath(join(_workerPath, "index.js")),
+				directory: resolvePath(_workerPath),
+				format: "modules",
+				moduleRoot: resolvePath(_workerPath),
+			},
+			[
+				{
+					type: "ESModule",
+					globs: ["**/*.js"],
+				},
+			]
+		);
+
+		const outfile = join(tmpdir(), `./bundledWorker-${Math.random()}.mjs`);
```

Comment:
If this is in a new temp dir, do you need the `Math.random()`?

Reply:
I guess not? But no harm in it. It's a pattern we use throughout Wrangler (and in particular, Pages Functions stuff).